File: flutter_app/lib/models/price_tick.dart
class PriceTick {
  final String symbol;
  final double price;
  final double previousClose;
  final double change;
  final double changePercent;
  final int timestamp;

  const PriceTick({
    required this.symbol,
    required this.price,
    required this.previousClose,
    required this.change,
    required this.changePercent,
    required this.timestamp,
  });

  factory PriceTick.fromJson(Map<String, dynamic> json) {
    return PriceTick(
      symbol: json['symbol'] as String,
      price: (json['price'] as num).toDouble(),
      previousClose: (json['previousClose'] as num).toDouble(),
      change: (json['change'] as num).toDouble(),
      changePercent: (json['changePercent'] as num).toDouble(),
      timestamp: json['timestamp'] as int,
    );
  }

  bool get isUp => change >= 0;
}

File: backend/GoldForexApi/Models/PriceTick.cs
namespace GoldForexApi.Models;

/// <summary>
/// A single live price update for one instrument (XAUUSD or GBPUSD).
/// This is what gets sent to the Flutter app, both over the websocket
/// and from the REST snapshot endpoint.
/// </summary>
public class PriceTick
{
    public string Symbol { get; set; } = string.Empty;      // "XAUUSD" or "GBPUSD"
    public decimal Price { get; set; }
    public decimal PreviousClose { get; set; }               // session open reference price
    public decimal Change { get; set; }
    public decimal ChangePercent { get; set; }
    public long Timestamp { get; set; }                      // ms since epoch (from Finnhub)
    public decimal Volume { get; set; }
}
